scipy/spatial/transform/_rigid_transform.pyx

```diff
@@ -0,0 +1,1943 @@
+              single transforms, the result is a single transform.
+            - Both ``p`` and ``q`` contain ``N`` transforms. In this case each
+              transform ``p[i]`` is composed with the corresponding transform
+              ``q[i]`` and the result contains ``N`` transforms.
+
+        Parameters
+        ----------
+        other : `RigidTransform` instance
+            Object containing the transforms to be composed with this one.
+
+        Returns
+        -------
+        `RigidTransform` instance
+            The composed transform.
+
+        Examples
+        --------
+        >>> from scipy.spatial.transform import RigidTransform as Tf
+        >>> from scipy.spatial.transform import Rotation as R
+        >>> import numpy as np
+
+        Compose two transforms:
+
+        >>> tf1 = Tf.from_translation([1, 0, 0])
+        >>> tf2 = Tf.from_translation([0, 1, 0])
+        >>> tf = tf1 * tf2
+        >>> tf.translation
+        array([1., 1., 0.])
+        >>> tf.single
+        True
+
+        When applied to a vector, the composition of two transforms is applied
+        in right-to-left order.
+
+        >>> t1, r1 = [1, 2, 3], R.from_euler('z', 60, degrees=True)
+        >>> t2, r2 = [0, 1, 0], R.from_euler('x', 30, degrees=True)
+        >>> tf1 = Tf.from_components(t1, r1)
+        >>> tf2 = Tf.from_components(t2, r2)
+        >>> tf = tf1 * tf2
+        >>> tf.apply([1, 0, 0])
+        array([0.6339746, 3.3660254, 3.       ])
+        >>> tf1.apply(tf2.apply([1, 0, 0]))
+        array([0.6339746, 3.3660254, 3.       ])
+
+        When at least one of the transforms is not single, the result is a stack
+        of transforms.
+
+        >>> tf1 = Tf.from_translation([1, 0, 0])
+        >>> tf2 = Tf.from_translation([[0, 2, 0], [0, 0, 3]])
+        >>> tf = tf1 * tf2
+        >>> tf.translation
+        array([[1., 2., 0.],
+               [1., 0., 3.]])
+        >>> tf.single
+        False
+        >>> len(tf)
+        2
+        """
+        len_self = len(self._matrix)
+        len_other = len(other._matrix)
+        if not(len_self == 1 or len_other == 1 or len_self == len_other):
+            raise ValueError("Expected equal number of transforms in both or a"
+                             "single transform in either object, got "
+                             f"{len_self} transforms in first and {len_other}"
+                             "transforms in second object.")
+
+        result = np.matmul(self._matrix, other._matrix)
+        if self._single and other._single:
+            result = result[0]
+        return self.__class__(result, copy=False)
+
+    @cython.embedsignature(True)
+    def __pow__(RigidTransform self, float n):
+        """Compose this transform with itself `n` times.
+
+        A rigid transform `p` can be raised to non-integer powers by applying
+        screw linear interpolation (ScLERP) to its underlying rotation and
```

Comment:
Minor detail, but I believe sclerp refers to interpolation between two transformations (poses). Technically this is not exactly the same. It is the step that is needed in sclerp because it is applied to the "difference" of the two transformations. Correct me if I a wrong here. Should be analogous to sleep for rotations

Reply:
I think maybe the clearest way is to say it's the interpolation between itself and the identity transformation?

Reply:
This might seem like an overcomplicated explanation. Is it not sufficient to say that we compute a fraction of the transformation? (Which is exactly what we do technically, just with exponential coordinates.)

Reply:
Updated, let me know what you think

Reply:
Sounds good!